crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,294 @@
+use super::Task;
+use anyhow::Result;
+use async_dropper_simple::{AsyncDrop, AsyncDropper};
+use async_trait::async_trait;
+use std::{fs::File, path::PathBuf, sync::Arc};
+use tokio::sync::RwLock;
+
+#[cfg(test)]
+use tests::{Child, InnerOctezNode};
+
+#[cfg(not(test))]
+use octez::OctezNode as InnerOctezNode;
+#[cfg(not(test))]
+use std::process::Child;
+
+#[derive(Clone)]
+pub struct OctezNodeConfig {
+    binary_path: PathBuf,
+    data_dir: PathBuf,
+    network: String,
+    rpc_endpoint: String,
+    log_file: PathBuf,
+    options: Vec<String>,
+}
+
+#[derive(Default)]
+pub struct OctezNodeConfigBuilder {
+    binary_path: Option<PathBuf>,
+    data_dir: Option<PathBuf>,
+    network: Option<String>,
+    rpc_endpoint: Option<String>,
+    log_file: Option<PathBuf>,
+    options: Option<Vec<String>>,
+}
+
+impl OctezNodeConfigBuilder {
+    pub fn new() -> Self {
+        OctezNodeConfigBuilder::default()
+    }
+
+    pub fn set_binary_path(&mut self, path: &str) -> &mut Self {
+        self.binary_path = Some(PathBuf::from(path));
+        self
+    }
+
+    pub fn set_data_dir(&mut self, path: &str) -> &mut Self {
+        self.data_dir = Some(PathBuf::from(path));
+        self
+    }
+
+    pub fn set_network(&mut self, network: &str) -> &mut Self {
+        self.network = Some(network.to_owned());
+        self
+    }
+
+    pub fn set_rpc_endpoint(&mut self, endpoint: &str) -> &mut Self {
+        self.rpc_endpoint = Some(endpoint.to_owned());
+        self
+    }
+
+    pub fn set_log_file(&mut self, path: &str) -> &mut Self {
+        self.log_file = Some(PathBuf::from(path));
+        self
+    }
+
+    pub fn set_run_options(&mut self, options: &[&str]) -> &mut Self {
+        self.options = Some(
+            options
+                .iter()
+                .map(|v| (*v).to_owned())
+                .collect::<Vec<String>>(),
+        );
+        self
+    }
+
+    pub fn build(&mut self) -> Result<OctezNodeConfig> {
+        Ok(OctezNodeConfig {
+            binary_path: self
+                .binary_path
+                .take()
+                .unwrap_or(PathBuf::from("octez-node")),
+            data_dir: self
+                .data_dir
+                .take()
+                .unwrap_or(PathBuf::from("/tmp/.octez-node")),
+            network: self.network.take().unwrap_or("sandbox".to_owned()),
+            rpc_endpoint: self
+                .rpc_endpoint
+                .take()
+                .unwrap_or("localhost:8732".to_owned()),
+            log_file: self
+                .log_file
+                .take()
+                .unwrap_or(PathBuf::from("/tmp/octez-node.log")),
```

Comment:
Perhaps we should use (tempfile crate)[https://docs.rs/tempfile/latest/tempfile/] for creating unique temps and cleaning them up on drop.

Reply:
Yup sure. I think we don't need to clean them up for now as the OS will do this at some point. I'm using `NamedTempFile` here so that we can keep track of the path.